### DISCUSSION TITLE
Support for React Native UI Libraries

### DISCUSSION THREAD

**Discussion by JonatanSalas** [vazco/uniforms#1142] [General]
Hi! At [BlackBox Vision](https://www.blackbox-vision.tech) we're working on adding support for React Native, we really appreciate the efforts you guys put into this library. 

As a first initiative, we are working on developing an adapter for React Native Paper, one of the most used packages for UI components in RN land. We're going to put efforts on RN only too, React Native Elements and other UI components libraries. 

Would really like to know if there's a chance of working together? And would like to know if there's a chance to merge our packages repository into the main repo when it's finished. **We're seeking to collaborate together.**

The repo adapter we're working on is here: https://github.com/BlackBoxVision/uniforms-react-native-paper

**Comment by radekmie** [comment]
Hey there, @JonatanSalas! We're always happy to see others building things on top of uniforms!

We've made a React Native theme some time ago (I think before v3), using [NativeBase](https://nativebase.io). It worked for us, but as we closed the project that used it, and we were no longer to dog food it ourselves, we've decided against publishing it.

To be honest, the same problem is there today. We're happy to help you out with the development, debugging, and promotion (we definitely should add a "featured" section on the website with community packages). But we don't have enough React Native projects as a company to be able to maintain it ourselves. That'd also mean, we'd rather not merge it into this repository, as it'd be hard for us to solve issues if there'll be any.

As the answer is rather vague, I propose a short call. You can find my email in my GitHub profile; we'll manage it.

**Reply by JonatanSalas** [reply]
Hey @radekmie! Thanks for your faster reply. I understand your point so there's no trouble, we can keep working on a separate repo. 

Would be really nice to have the library shown on the docs site! 

My initial proposal was related to joining forces and having some sort of communication with your team to be aligned about versioning and possible breaking changes. I think we can have a short call, maybe I will reach you so we can meet and talk a little. 

Thank you!

**Reply by radekmie** [reply]
@JonatanSalas Are you still interested in a short talk? _(I know we didn't update the docs yet._)
